### ISSUE TITLE
Roku Limited Mode

### ISSUE THREAD

**Issue by tcraigtyler** [home-assistant/home-assistant.github.io#36240]
### Feedback

Roku seems to have added a Limited option to their "Control by mobile apps" that prevents the typical access I had previously. In the devices settings I was seeing an error "ECP command not allowed in limited mode". When I deleted the integration and tried to re-add it the device was not found when I entered the IP address. 

After looking around in the Roku menus I found "Control by mobile apps" and changed it to "Enabled", re-added the integration, and now I have the same access as before.

[Previous documentation](https://community.roku.com/t5/Features-settings-updates/Can-t-cast-or-control-my-Roku-via-my-phone/m-p/996952/highlight/true#M49407) of "Control by mobile apps" as of two months ago
[Current documentation](https://support.roku.com/en-gb/article/217288467#section-2) of "Control by mobile apps"

It would be good to mention changing "Control by mobile apps" to the setup instructions.

### URL

https://www.home-assistant.io/integrations/roku/

### Version

2024.12.1

**Comment by bob454522** [comment]
I can confirm the same (43" TCL roku tv)  -  I was getting the same error on the roku integration and my on-off command was not working, I also noticed that the TV did a auto firmware update overnight (today, dec13 2024) - so this update/reboot must be related.

fix was the same as OP - on the TV i went to: settings -> Advanced Settings -> and found "Control by mobile apps" and changed it to "Enabled", and then the HA integration Immediately showed as good / started working again (I didn't have to reload it or do anything else).

Since there is no way to disable automatic firmware updates, for now I've added a firewall rule on my router blocking Internet access to the Roku TV (but local IPs still allowed ofcouse).  im sure the tv will stop working or give some error due to this FW rule at some point..

**Comment by astone123** [comment]
I just ran into this today too, going into settings on my Roku TV and changing "Settings > System > Advanced Settings > Control by mobile apps" to "Permissive" allowed Home Assistant to communicate with it again.

**Comment by diamantopoulos12** [comment]
Thanks for this post!  All 3 of my rokus updated automatically the other day and blocked homeassistant from connecting.  Enabling this fixed it.

**Comment by quincarter** [comment]
Yep - can confirm this is the fix. Thanks all. I found this as the top result when searching.
